charts/keycloakx/README.md

```diff
@@ -0,0 +1,628 @@
+See example for Google Cloud Proxy or default affinity configuration in `values.yaml`.
+
+### JVM Settings
+
+Keycloak sets the following system properties by default:
+`-Djava.net.preferIPv4Stack=true -Djboss.modules.system.pkgs=$JBOSS_MODULES_SYSTEM_PKGS -Djava.awt.headless=true`
+
+You can override these by setting the `JAVA_OPTS` environment variable.
+Make sure you configure container support.
+This allows you to only configure memory using Kubernetes resources and the JVM will automatically adapt.
+
+```yaml
+extraEnv: |
+  - name: JAVA_OPTS
+    value: >-
+      -XX:+UseContainerSupport
+      -XX:MaxRAMPercentage=50.0
+      -Djava.net.preferIPv4Stack=true
+      -Djava.awt.headless=true
+```
+
+### Database Setup
+
+By default, Bitnami's [PostgreSQL](https://github.com/bitnami/charts/tree/master/bitnami/postgresql) chart is deployed and used as database.
+Please refer to this chart for additional PostgreSQL configuration options.
+
+#### Using an External Database
+
+The Keycloak Docker image supports various database types.
+Configuration happens in a generic manner.
+
+##### Using a Secret Managed by the Chart
+
+The following examples uses a PostgreSQL database with a secret that is managed by the Helm chart.
+
+```yaml
+postgresql:
+  # Disable PostgreSQL dependency
+  enabled: false
+
+extraEnv: |
+  - name: KC_DB
+    value: postgres
+  - name: KC_DB_URL_HOST
+    value: mypostgres
+  - name: KC_DB_URL_PORT
+    value: "5432"
+  - name: KC_DB_DATABASE
+    value: mydb
+
+extraEnvFrom: |
+  - secretRef:
+      name: '{{ include "keycloak.fullname" . }}-db'
+
+secrets:
+  db:
+    stringData:
+      KC_DB_USERNAME: '{{ .Values.dbUser }}'
+      KC_DB_PASSWORD: '{{ .Values.dbPassword }}'
+```
+
+`dbUser` and `dbPassword` are custom values you'd then specify on the commandline using `--set-string`.
+
+##### Using an Existing Secret
+
+The following examples uses a PostgreSQL database with a secret.
+Username and password are mounted as files.
+
+```yaml
+postgresql:
+  # Disable PostgreSQL dependency
+  enabled: false
+
+extraEnv: |
+  - name: KC_DB
+    value: postgres
+  - name: KC_DB_URL_HOST
+    value: mypostgres
+  - name: KC_DB_URL_PORT
+    value: "5432"
+  - name: KC_DB_DATABASE
+    value: mydb
+  - name: KC_DB_USERNAME_FILE
+    value: /secrets/db-creds/user
+  - name: KC_DB_PASSWORD_FILE
+    value: /secrets/db-creds/password
+
+extraVolumeMounts: |
+  - name: db-creds
+    mountPath: /secrets/db-creds
+    readOnly: true
+
+extraVolumes: |
+  - name: db-creds
+    secret:
+      secretName: keycloak-db-creds
+```
+
+### Creating a Keycloak Admin User
+
+The Keycloak-X Docker image supports creating an initial admin user.
+It must be configured via environment variables:
+
+* `KC_DB_USERNAME` or `KC_DB_USERNAME_FILE`
+* `KC_DB_PASSWORD` or `KC_DB_PASSWORD_FILE`
+
+Please refer to the section on database configuration for how to configure a secret for this.
+
+### High Availability and Clustering
+
+TODO Revise
```

Comment:
Here is still a todo. Should this be resolved before merging?

Reply:
Revised the HA section